update wording of Nomad Client Counting FAQ to reflect the new Integration (#28685)
hashicorp/vault: website/content/partials/faq/client-count/computing-clients.mdx

### Does the Nomad-Vault integration affect client counts? ((#nomad))

**Yes**.

The [Nomad Vault integration](/nomad/docs/integrations/vault-integration)
uses either Workload Identity (JWT) or token roles for client count:

> 	- Workload Identity client counts use based on the
> 	  [`user_claim`](/vault/api-docs/auth/jwt#user_claim) field. The recommended default is `nomad_job`,
>     which results in 1 client per Nomad job.

Reviewer: ```suggestion
	- Vault bases Workload Identity client counts on the
	  [`user_claim`](/vault/api-docs/auth/jwt#user_claim) field. The
	  recommended default is `nomad_job`, which results in 1 client per
	  Nomad job.
```
Fixing copy/paste/edit error

> 	- Nomad has deprecated token roles and they will be removed in v1.10. Client counts for legacy
> 	  token roles are 1 non-entity client per unique each policy combination.

Reviewer: ```suggestion
	- Nomad deprecated token roles and will remove the feature in v1.10. Client
	  counts for legacy token roles treat each unique policy combination as 1
		non-entity client.
```
Style correction: write in active voice, fix some copy/paste/edit errors

**Maybe**.

[Nomad Vault integration](/nomad/docs/integrations/vault-integration#token-role-based-integration)

> uses either Workload Identity (JWT) or token roles (deprecated and removed in Nomad 1.10).
> Client counts with Workload Identity will be based on the [`user_claim`](/vault/api-docs/auth/jwt#user_claim),
> usually 1 client per Nomad job. With the legacy token role integration, there is 1
> non entity client per unique combination of policies requested by Nomad jobs.

Reviewer: ```suggestion
uses either Workload Identity (JWT) or token roles for client count:

- Workload Identity client counts use the
  [`user_claim`](/vault/api-docs/auth/jwt#user_claim) field. By default, the
  count is 1 client per Nomad job.
- Nomad deprecated and removed token roles in v1.10. Client counts for legacy
  token roles use  1 non-entity client per unique (Nomad job + policy)
  combination.
```

Style correction: write in active voice. Make key elements easy to scan
